### PR TITLE
DD-1171: add SHA1 manifest if not present

### PR DESCRIPTION
**Pull request by jo-pol**
Fixes DD-1171: add SHA1 manifest if not present

# Description of changes
* add SHA1 manifest if not present
* update tagmanifest
* moved a static method from the `FileEment` class to the new `ManifestHelper` class

# How to test

# Related PRs

(Add links)

*

# Notify

@DANS-KNAW/dataversedans

### CODE REVIEW

File: src/main/java/nl/knaw/dans/ingest/core/service/ManifestHelper.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                else return super.visitFile(path, attrs);
+            }
+        };
+        Files.walkFileTree(bagRootDir, tagVisitor);
+        bag.getTagManifests().clear();
+        bag.getTagManifests().addAll(tagFilesMap.keySet());
+        ManifestWriter.writeTagManifests(bag.getTagManifests(), PathUtils.getBagitDir(bag), bagRootDir, bag.getFileEncoding());
+    }
+
+    static public Map<Path, String> getFilePathToSha1(Deposit deposit) {
```
Comment (jo-pol) [inline src/main/java/nl/knaw/dans/ingest/core/service/ManifestHelper.java:88]: This method is moved here from `FileElement` as is. I'm hesitating to change the argument to a bag.

File: src/main/java/nl/knaw/dans/ingest/core/service/DepositIngestTask.java [hunk 2]
```diff
@@ -221,6 +221,13 @@ void validateDeposit() {
                     "Bag was not valid according to Profile Version %s. Violations: %s",
                     result.getProfileVersion(), violations)
                 );
+            } else {
```
Comment (janvanmansum) [inline src/main/java/nl/knaw/dans/ingest/core/service/DepositIngestTask.java:224]: Avoid negatives in if-conditions. Better would be `if(result.isCompliant) ensureSha1ManifestPresent() else ...`.

Better still is to always `ensureSha1ManifestPresent` before calling the validator, so you have a doublecheck on your (possibly added) SHA1-manifest.

File: src/main/java/nl/knaw/dans/ingest/core/service/ManifestHelper.java [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static gov.loc.repository.bagit.hash.StandardSupportedAlgorithms.SHA1;
+
+public class ManifestHelper {
+
+    static public void addSha1File(Bag bag) throws NoSuchAlgorithmException, IOException {
```
Comment (janvanmansum) [inline src/main/java/nl/knaw/dans/ingest/core/service/ManifestHelper.java:42]:
```suggestion
    static public void ensureSha1ManifestPresent(Bag bag) throws NoSuchAlgorithmException, IOException {
```
to better cover what this method does.

File: src/main/java/nl/knaw/dans/ingest/core/service/ManifestHelper.java (hunk 1 above)
Comment (janvanmansum) [reply src/main/java/nl/knaw/dans/ingest/core/service/ManifestHelper.java:88]: Deposit is used to find the bag directory in line 97, if you can do that without the Deposit object, I guess you could refactor this method.

**Review by janvanmansum — CHANGES_REQUESTED** [review]
See my comments.

File: src/main/java/nl/knaw/dans/ingest/core/service/DepositIngestTask.java (hunk 2 above)
Comment (jo-pol) [reply src/main/java/nl/knaw/dans/ingest/core/service/DepositIngestTask.java:224]: Not sure I agree. You would repeat an expensive calculation and you don't know for sure any more whether violations are about the original bag or the altered bag.

**Pull request merged**